eventually-postgres/src/subscriber.rs

```diff
@@ -23,20 +23,26 @@ use tokio_postgres::AsyncMessage;
 
 const DEFAULT_BROADCAST_CHANNEL_SIZE: usize = 128;
 
-/// Alias type for a `Result` having [`DeserializeError`] as error type.alloc
+/// Alias type for a `Result` having [`SubscriberError`] as error type.alloc
 ///
-/// [`DeserializeError`]: struct.DeserializeError.html
-pub type Result<T> = std::result::Result<T, DeserializeError>;
+/// [`SubscriberError`]: struct.SubscriberError.html
+pub type Result<T> = std::result::Result<T, SubscriberError>;
 
 /// Error returned by the `TryStream` on [`subscribe_all`]
-/// when deserializing payloads coming from Postgres' `LISTEN`
-/// asynchronous notifications.
 ///
 /// [`subscribe_all`]: struct.EventSubscriber.html#method.subscribe_all
 #[derive(Debug, Clone, thiserror::Error)]
-#[error("failed to deserialize notification payload from JSON: {message}")]
-pub struct DeserializeError {
-    message: String,
+pub enum SubscriberError {
+    /// Error variant returned when deserializing payloads coming from Postgres' `LISTEN`
+    /// asynchronous notifications.
+    #[error("failed to deserialize notification payload from JSON: {0}")]
+    DeserializeError(String),
+
+    /// Error variant returned when the connection, used for `LISTEN` asynchronous notifications
+    /// gets dropped. Currently the subscriber cannot recover from this error and a new one should
+    /// be created.
+    #[error("postgres connection error: ${0}")]
+    ConnectionError(String),
```

Comment:
Can we use `tokio_postgres::Error` here instead of `String`?

```suggestion
    ConnectionError(#[source] tokio_postgres),
```

Reply:
`tokio_postgres` error is not `Clone`, and it needs to be `Clone` to send over the channel, as far as I can tell

Reply:
Oh yes, indeed. Another "smell" of using a channel to send these values around 😅